sqlserver/datadog_checks/sqlserver/sqlserver.py

```diff
@@ -582,7 +582,12 @@ def check(self, _):
             if self.autodiscovery and self.autodiscovery_db_service_check:
                 for db_name in self.databases:
                     if db_name != self.connection.DEFAULT_DATABASE:
-                        self.connection.check_database_conns(db_name)
+                        try:
+                            self.connection.check_database_conns(db_name)
+                        except Exception as e:
```

Comment:
While it might make sense to have a wrapper here regardless for actually unexpected situations, it looks like the behavior is to intentionally raise an exception:

https://github.com/DataDog/integrations-core/blob/502b6d891ec86e44bb5a30ae9e150d4f95a0ced3/sqlserver/datadog_checks/sqlserver/connection.py#L190

Is this something we want to keep?

Reply:
Good question. I have to make educated guesses here but I think raising the Exception on service check was probably meant to abort the check when the integration couldn't connect to the instance (main database). It did make sense in that context but it started to make less sense with secondary/auto-discovered databases. 

There's already a notion of `is_default` being passed around so maybe we should only raise the exception when `is_default == true`?

Reply:
Updated to only do the `raise_from(SQLConnectionError(message), None)` when `is_default` is `True`.

Reply:
> I think this might be due for a refactor so that we're separating the concerns of the service check and opening the connection (and the error flow for autodiscovery vs. default db), but that can be a discussion for another time.

Yes, I agree a refactor would help here. We should probably plan an exercise of reviewing potential failure modes in all the database integrations and see how we should update things now that we have dbm functionality. Related ticket: [DBM-730](https://datadoghq.atlassian.net/browse/DBM-730)